rfcs/0009-enclave.md

```diff
@@ -31,6 +31,15 @@ Secure Enclaves (such as SGX) provide a layer of memory and computation protecti
     * The sign request contains the entry ID for the new vote
     * Enclave verifies that new vote's entry ID is on the correct branch (following the rules #1 and #2 above)
 
+
+## Branch determination
+
+Due to the fact that the enclave cannot process PoH, it has no direct knowledge of branch history of a submitted validator vote. Each enclave should be initiated with the current *active set* of public keys. A validator should submit its current vote along with the votes of the active set (including itself) that it observed in the slot of its previous vote. In this way, the enclave can surmise the votes accompanying the validator's previous vote and thus the branch being voted on. This is not possible for the validator's initial submitted vote, as it will not have a 'previous' slot to reference. To account for this, a short voting freeze should apply until the second vote is submitted containing the votes within the active set, along with it's own vote, at the height of the initial vote.
```

Comment:
Is it correct to assume that the validator submits the PoH entries generated between the current vote and last vote (in addition to the votes of the active set)? This will be used to compute the branch that the validator is trying to vote on.

Reply:
Is Voting Freeze enforced by the enclave? How does signing process differ during a freeze?

Reply:
@pgarg66
1. yes.  thats correct
2. maybe?  the network just doesn't allow the vote to count so no rewards are generated